### ISSUE TITLE
[rule]: no-ternary should not be an error by default

### ISSUE THREAD

**Issue by aparajita** [oxc-project/oxc#2768]
Really? Making all ternary expressions an error by default? Please think through this a little more deeply.

**Comment by Boshen** [comment]
`no-ternary` is marked as a style rule and is not turned on by default, and I cannot reproduce this by running `pnpx oxlint@latest test.js` and a test.js `a ? b : c`.

**Comment by aparajita** [comment]
Thanks for responding, I found it hard to believe this was intentional. Here is how I was invoking oxlint@0.2.14:

```sh
oxlint --deny-warnings \
-D correctness \
-D restriction \
-A no-console -A no-void \
-D pedantic -A prefer-query-selector \
-D style
-A explicit-length-check \
-A filename-case \
-A no-null \
-A no-nested-ternary \
-A switch-case-braces \
-A empty-brace-spaces \
-A no-mixed-operators \
-A no-continue \
-A no-negated-condition \
-A prefer-string-replace-all \
-A catch-error-name \
-A array-type \
-A no-process-exit \
-D suspicious
```

**Comment by Boshen** [comment]
You supplied `-D style`, this will turn on all stylistic rules. My recommendation is to turn the whole `style` category off and pick the ones you need, just like how we used to do things with eslint 😅

**Comment by aparajita** [comment]
Got it, thanks.
